chatper08/chapter08.adoc

```diff
@@ -0,0 +1,654 @@
+|===
+
+Below is an example of configuring a circuit breaker for a service method using the `@CircuitBreaker` annotation:
+
+[source,java]
+----
+@CircuitBreaker(
+    requestVolumeThreshold = 10,
+    failureRatio = 0.5,
+    delay = 5000,
+    successThreshold = 2,
+    failOn = RuntimeException.class
+)
+public String getProduct(Long id) {
+    // Logic to call the product details service
+    if (Math.random() > 0.7) {
+        throw new RuntimeException("Simulated service failure");
+    }
+    return productRepository.findProductById(id);
+}
+----
+
+In the above code, the circuit breaker opens if 50% of requests fail (`failureRatio = 0.5`) after at least 10 requests (`requestVolumeThreshold = 10`). It remains open for 5 seconds (`delay = 5000`) and transitions to the "half-open" state to test recovery. Two consecutive successful requests (`successThreshold = 2`) in the "half-open" state close the circuit breaker.
+
+==== Best Practices for Circuit Breaker
+
+- **Set Realistic Failure Ratios and Thresholds:** Tailor parameters to your services' expected load and failure behavior.
+- **Monitor Metrics:** Use MicroProfile Metrics to monitor circuit breaker state transitions.
+- **Combine with Other Strategies:** Use circuit breakers alongside retries and timeouts for a robust fault tolerance setup.
+
+=== Setting Timeouts
+
+Timeouts are an essential fault tolerance strategy to prevent long-running operations from consuming resources indefinitely. Slow or unresponsive services can degrade overall system performance and reliability in a microservices architecture. The `@Timeout` annotation provided by MicroProfile Fault Tolerance allows you to define a maximum duration for a method to complete, ensuring that system resources remain available for other tasks.
+
+==== Why Use Timeouts?
+
+In distributed systems, slow responses from downstream services can cascade through the system, leading to resource contention and degraded performance. Timeouts allow you to:
+- Abort operations that exceed acceptable time limits.
+- Free system resources for other operations.
+- Trigger alternative strategies, such as fallbacks, to maintain functionality.
+
+[source,java]
+----
+import org.eclipse.microprofile.faulttolerance.Timeout;
+
+public class PaymentService {
+
+    @Timeout(1000)
+    public String processPayment(PaymentDetails paymentDetails) {
+        // Simulate a long-running process
+        try {
+            Thread.sleep(1500);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException("Processing interrupted");
+        }
+        return "Payment processed successfully.";
+    }
+}
+----
+
+In this example:
+- The `@Timeout(1000)` annotation specifies that the `processPayment` method must complete within 1000 milliseconds (1 second).
+- If the execution exceeds this time, a `TimeoutException` will be thrown, and the process will terminate.
```

Comment:
Can it really guarantee this?  Or is it subject to the process being interruptible?  If the latter, the wording should be softened to reflect that. For example,

```suggestion
- If the execution exceeds this time, an attempt is made to interrupt the process, and a `TimeoutException` will be thrown when the process terminates.
```